qp_shotgun/kneaddata/kneaddata.py

```diff
@@ -0,0 +1,287 @@
+                             '%s' % run_prefix)
+
+        if rev_fp is None:
+            samples.append((run_prefix, sn_by_rp[run_prefix], fwd_fp, None))
+        else:
+            rev_fn = basename(rev_fp)
+            # if we have reverse reads, make sure the matching pair also
+            # matches the run prefix:
+            if not rev_fn.startswith(run_prefix):
+                raise ValueError('Reverse read does not match this run prefix.'
+                                 '\nRun prefix: %s\nForward read: %s\n'
+                                 'Reverse read: %s\n' %
+                                 (run_prefix, fwd_fn, rev_fn))
+
+            samples.append((run_prefix, sn_by_rp[run_prefix], fwd_fp,
+                            rev_fp))
+
+        used_prefixes.add(run_prefix)
+
+    return(samples)
+
+
+def format_kneaddata_params(parameters):
+
+    params = []
+
+    for param in sorted(parameters):
+        value = parameters[param]
+
+        if value is True:
+            params.append('--%s' % param)
+            continue
+        elif value is False:
+            continue
+        elif value:
+            params.append('--%s %s' % (param, value))
+            continue
+
+    param_string = ' '.join(params)
+
+    return(param_string)
+
+
+def generate_kneaddata_commands(forward_seqs, reverse_seqs, map_file,
+                                out_dir, parameters):
+    """Generates the KneadData commands
+
+    Parameters
+    ----------
+    forward_seqs : list of str
+        The list of forward seqs filepaths
+    reverse_seqs : list of str
+        The list of reverse seqs filepaths
+    map_file : str
+        The path to the mapping file
+    out_dir : str
+        The job output directory
+    parameters : dict
+        The command's parameters, keyed by parameter name
+
+    Returns
+    -------
+    cmds: list of str
+        The KneadData commands
+    samples: list of tup
+        list of 3-tuples with run prefix, sample name, fwd read fp, rev read fp
+
+    Raises
+    ------
+    ValueError
+        If the rev is not an empty list and the same length than fwd seqs
+        The prefixes of the run_prefix don't match the file names
+
+    Notes
+    -----
+    Currently this is requiring matched pairs in the make_read_pairs_per_sample
+    step but implicitly allowing empty reverse reads in the actual command
+    generation. This behavior may allow support of situations with empty
+    reverse reads in some samples, for example after trimming and QC.
+    """
+    # making sure the forward and reverse reads are in the same order
+
+    # we match filenames, samples, and run prefixes
+    samples = make_read_pairs_per_sample(forward_seqs, reverse_seqs, map_file)
+
+    cmds = []
+
+    param_string = format_kneaddata_params(parameters)
+    prefixes = []
+    for run_prefix, sample, f_fp, r_fp in samples:
+        prefixes.append(run_prefix)
+        if r_fp is None:
```

Comment:
Minor, but I think this if/else block can be reduced to:
```python
r_fp_str = ' --input "%s"' % r_fp if r_fp is not None else ""
cmds.append('kneaddata --input "%s"%s --output "%s" --output-prefix "%s" %s'
            % (f_fp, r_fp_str, join(out_dir, run_prefix), run_prefix, param_string))
```
to avoid some code duplication